subsys/jaxrs/src/main/java/org/commonjava/indy/bind/jaxrs/HeaderDebugger.java

```diff
@@ -35,14 +36,16 @@
 
     private final Logger logger = LoggerFactory.getLogger( getClass() );
 
-    private final HttpHandler handler;
+    private HttpHandler handler;
+
+    public HeaderDebugger() {}
 
     public HeaderDebugger( final HttpHandler handler )
     {
         this.handler = handler;
     }
 
-    public static final class Wrapper
+    public class Wrapper
```

Comment:
Okay, I've been looking online, and I can't find any place where static classes are treated as singletons. Static fields definitely are, and static methods are object-less...but static classes are just inner classes that don't maintain a reference to the object / class that contains them. With non-static inner classes, you can use Outer.this.foo() references (where Outer is the outer classname). With this change, you're creating a Wrapper instance that contains a reference to the outer HeaderDebugger you used to instantiate it...so you could set the wrap method to return HeaderDebugger.this.

I will admit that the inner-outer split on this class is very strange, though. Normally, the outer class would have been the Wrapper class, and the inner class would have been `public static final class HeaderDebugger`...so I constructed this in a weird way.

I think more importantly here, we don't need to use this class at all. I was using it temporarily to help debug something I was implementing, and left it in "just in case"...which is the root of an awful lot of evil in the world of code (in my experience).

I really wonder if your additional `Option` configurations actually made the difference in performance?

Reply:
I have try it with commented out method where we add that class and also i have compare both static and without static wrapper and results for performance is slightly better for non-static or commented out wrapper class but most important and only difference is that when i use static wrapper class http test tool is not finishing to end , it is failing in more than 60% of cases with time-outs.
BTW i agree on commenting it out , maybe for now is best solution.
Also i have try only OPTIONS configs for keep-alive settings and surpassingly it is with better performance for keep-alive to be false and not true .

Reply:
Ok then, we should follow the data.